### PR TITLE
Allow posting comments with authentication

### PR DESCRIPTION
**Pull request by trevanhetzel**
With the use of cookie authentication, this allows you to `POST` to the WP REST API to create a comment.

To use this in your theme, you simply need to add a `nonce` attribute to the `<wordpress-api>` component.

`<wordpress-api nonce="<?php echo wp_create_nonce( 'wp_rest' ); ?>"></wordpress-api>`

To clean this up, we'd want to move the post methods to the the base API class, isntead of it living solely on the comments interactor. Also need to move the query selector for the nonce out into a prop and add a nil check. This PR is pretty basic just to prove the concept of posting to the API with auth.

### CODE REVIEW

File: src/components/wordpress-api/interactors/comments.ts [hunk 1]
```diff
@@ -79,6 +79,14 @@ export class Comments {
   async getBySlug(slug: string): Promise<any> {
     return this.db.comments.where("slug").equals(slug).first();
   }
+
+  async create(payload: object) {
+    const result = await this.api.post(payload);
+
+    // await this.db.comments.put(result);
```
Comment (trevanhetzel) [inline src/components/wordpress-api/interactors/comments.ts:86]: This is where we'd update indexedb

File: package.json [hunk 2]
```diff
@@ -19,7 +19,8 @@
   },
   "dependencies": {
     "@stencil/core": "next",
-    "dexie": "^2.0.1"
+    "dexie": "^2.0.1",
+    "request": "^2.83.0"
```
Comment (splitinfinities) [inline package.json:23]: I'm going to remove this dependency after we merge this as a heads up!

File: src/components/wordpress-api/interactors/comments.ts (hunk 1 above)
Comment (splitinfinities) [reply src/components/wordpress-api/interactors/comments.ts:86]: Nice!

File: src/components/wordpress-api/interactors/comments.ts [hunk 3]
```diff
@@ -101,6 +109,25 @@ class CommentsAPI extends BaseAPI {
 
     return this.fetch(`?per_page=${args.limit}&offset=${args.offset}`);
   }
+
+  async post(payload) {
```
Comment (splitinfinities) [inline src/components/wordpress-api/interactors/comments.ts:113]: Super cool.

**Review by splitinfinities — APPROVED** [review]
Looks so awesome! I'm merging. I'll do a little bit of clean up then cut a new version. 🎉

**Pull request merged**